src/components/FileItem.tsx

```diff
@@ -117,9 +117,17 @@ export interface IFileItemProps {
    */
   selected?: boolean;
   /**
-   * Callback to select the file
+   * Callback to toggle select/deselect the file
    */
-  selectFile?: (file: Git.IStatusFile | null) => void;
+  toggleFile?: (file: Git.IStatusFile) => void;
+  /**
+   * Callback to handle shift-click on the file
+   */
+  handleShiftClick?: (file: Git.IStatusFile) => void;
+  /**
+   * Callback to replace the selected files
+   */
+  selectOnlyOneFile?: (file: Git.IStatusFile) => void;
```

Comment:
Could you use a single callback like:

```
setSelection(file: Git.IStatusFile, options: {singleton?: boolean, group?: boolean})
```

`toogleFile` :arrow_right: `setSelection(file)`
`handleShiftClick` :arrow_right: `setSelection(file, {group: true})`
`selectOnlyOneFile` :arrow_right: `setSelection(file, {singleton: true})`